test/AsmResolver.Workspaces.DotNet.Tests/AssemblyTraversalTest.cs

```diff
@@ -0,0 +1,268 @@
+        {
+            var parameterDefinitions = GetAllMembers<ParameterDefinition>(_fixture.AllAssemblies, TableIndex.Param);
+            TraverseObjects(parameterDefinitions);
+        }
+
+        [Fact]
+        public void InterfaceImplCheck()
+        {
+            var interfaceImplementations = GetAllMembers<InterfaceImplementation>(_fixture.AllAssemblies, TableIndex.InterfaceImpl);
+            TraverseObjects(interfaceImplementations);
+        }
+
+        [Fact]
+        public void MemberRefCheck()
+        {
+            var memberReferences = GetAllMembers<MemberReference>(_fixture.AllAssemblies, TableIndex.MemberRef);
+            TraverseObjects(memberReferences);
+        }
+
+        [Fact]
+        public void CustomAttributeCheck()
+        {
+            var customAttributes = GetAllMembers<CustomAttribute>(_fixture.AllAssemblies, TableIndex.CustomAttribute);
+            TraverseObjects(customAttributes);
+        }
+
+        [Fact]
+        public void DeclSecurityCheck()
+        {
+            var securityDeclarations = GetAllMembers<SecurityDeclaration>(_fixture.AllAssemblies, TableIndex.DeclSecurity);
+            TraverseObjects(securityDeclarations);
+        }
+
+
+        //TODO: Some standalone signatures are used for assembly building but not afterwards so some filtering is required
+        /*[Fact]
+        public void StandAloneSigCheck()
+        {
+            var standAloneSignatures = GetAllMembers<StandAloneSignature>(_fixture.AllAssemblies, TableIndex.StandAloneSig)
+            TraverseObjects(standAloneSignatures);
+        }*/
+
+
+        [Fact]
+        public void EventCheck()
+        {
+            var eventDefinitions = GetAllMembers<EventDefinition>(_fixture.AllAssemblies, TableIndex.Event);
+            TraverseObjects(eventDefinitions);
+        }
+
+        [Fact]
+        public void PropertyCheck()
+        {
+            var propertyDefinitions = GetAllMembers<PropertyDefinition>(_fixture.AllAssemblies, TableIndex.Property);
+            TraverseObjects(propertyDefinitions);
+        }
+
+        [Fact]
+        public void ModuleRefCheck()
+        {
+            var moduleReferences = GetAllMembers<ModuleReference>(_fixture.AllAssemblies, TableIndex.ModuleRef);
+            TraverseObjects(moduleReferences);
+        }
+
+        [Fact]
+        public void TypeSpecCheck()
+        {
+            var typeSpecifications = GetAllMembers<TypeSpecification>(_fixture.AllAssemblies, TableIndex.TypeSpec);
+            TraverseObjects(typeSpecifications);
+        }
+
+        [Fact]
+        public void AssemblyCheck()
+        {
+            var assemblyDefinitions = GetAllMembers<AssemblyDefinition>(_fixture.AllAssemblies, TableIndex.Assembly);
+            TraverseObjects(assemblyDefinitions);
+        }
+
+        [Fact]
+        public void FileCheck()
+        {
+            /* Due to not having any assembly with file references we will use custom generated assembly
```

Comment:
We do have them actually. Any multi-module assembly uses file references to refer to the secondary modules. In AsmResolver.DotNet.Tests' resources, `Manifest.exe` refers to `MyModel.netmodule` in the file references. Although perhaps this test is better than the pre-compiled one.